File: shoppingCart/src/main/java/com/shop/top/shoppingcart/models/Order.java
package com.shop.top.shoppingcart.models;

import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.time.LocalDate;
import java.util.List;

@Entity
@Setter
@Getter
public class Order {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "order_id", nullable = false)
    private Long orderId;

    @Column("user_id")
    private Long userId;

    @Column("user_name")
    private String userName;

    @Column(name = "created_date")
    private LocalDate createdDate;

    private String status;

    private float amount;

    @OneToMany(fetch = FetchType.EAGER, cascade = CascadeType.ALL)
    @JoinColumn("order_id")
    private List<OrderDetail> orderDetails;
}

File: Top-Shop-UI/src/app/modals/order.ts
import { OrderDetail } from './order-detail';

export class Order {

    id? : number;
    userId? : string;
    userName? : string;
    createdDate? : Date;
    status? : string;
    amount? : number;
    orderDetails? : OrderDetail[];

    constructor(userId : string, userName : string, createdDate : Date, 
                status : string, amount : number, orderDetails : OrderDetail[]){
        this.userId = userId;
        this.userName = userName;
        this.createdDate = createdDate;
        this.status = status;
        this.amount = amount;
        this.orderDetails = orderDetails;
    }

}